### PR TITLE
Implement POST /eth-signature/token

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related #723

### CODE REVIEW

**Comment by mohammadranjbarz** [comment]
@kristoferlund  I wasn't sure about endpoint address and the controller to put that code, so let me know if you think  it's better to change it

**Review by kristoferlund — CHANGES_REQUESTED** [review]
Looks good! 

Perhaps name of the endpoint should be `eth-signature/refresh`?

I'd like to see some more tests:
1. Make request with expired token fails
2. Refresh token
3. Make same request as 1, now it works

**Comment by mohammadranjbarz** [comment]
> Looks good!
> 
> Perhaps name of the endpoint should be `eth-signature/refresh`?
> 
> I'd like to see some more tests:
> 
> 1. Make request with expired token fails
> 2. Refresh token
> 3. Make same request as 1, now it works

Thanks for your feedback, I'll do them

**Comment by mohammadranjbarz** [comment]
> Looks good!
> 
> Perhaps name of the endpoint should be `eth-signature/refresh`?
> 
> I'd like to see some more tests:
> 
> 1. Make request with expired token fails
> 2. Refresh token
> 3. Make same request as 1, now it works

@kristoferlund 
I made all changes that you wanted and added some more test cases
but I get this error in ci/cd but the error lines doesn't exist in the codes and I can build locally, can you help me on that

https://github.com/givepraise/praise/actions/runs/5199598195/jobs/9377208506

Screen Shot 1402-03-17 at 15 30 54

**Comment by kristoferlund** [comment]
@mohammadranjbarz I think the user object in `LoginResponseDto` was added recently by @nebs-dev, have you merged  with latest  changes from `main`?

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz I think the user object in `LoginResponseDto` was added recently by @nebs-dev, have you merged with latest changes from `main`?

@kristoferlund  I just merged `main` to my branch and it got fixed, thank you

**Review by kristoferlund — CHANGES_REQUESTED** [review]
Looks good, only one small fix needed. We should not use 500 response (only when something is broken).

File: packages/api/src/auth/eth-signature.service.ts [hunk 1]
```diff
@@ -87,17 +86,107 @@ export class EthSignatureService {
...
+   *
+   * @param token String
+   * @param hostname String
+   * @returns LoginResponse
+   */
+  async generateTokensByRefreshToken(
+    token: string,
+    hostname: string,
+  ): Promise<LoginResponseDto> {
+    const tokenPayload = this.jwtService.verify(token, {
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply packages/api/src/auth/eth-signature.service.ts]: @kristoferlund  Done

**Review by kristoferlund — APPROVED** [review]
Great, thanks! Merging.

**Pull request merged**